torch/csrc/Size.cpp

```diff
@@ -148,8 +148,33 @@ static PyObject *THPSize_numel(THPSize *self)
   END_HANDLE_TH_ERRORS
 }
 
+static PyObject *THPSize_reduce(THPSize *self)
+{
+  HANDLE_TH_ERRORS
+  auto ret = THPObjectPtr{PyTuple_New(2)};
+  if (!ret) throw python_error();
+
+  auto module = THPObjectPtr(PyImport_ImportModule("torch"));
+  if (!module) throw python_error();
+
+  THPObjectPtr obj(PyObject_GetAttrString(module, "Size"));
```

Comment:
I think `obj` is just `THPSizeType`. You'll need to cast to a `PyObject*` and increment the reference count.